### PR TITLE
VxAdmin: Redesign smart card programming flow

### LINKED ISSUE #5353: Update card programming UX
https://votingworks.slack.com/archives/CJU9MSC6S/p1725366158625429?thread_ts=1725055786.379879&cid=CJU9MSC6S

Also related: https://github.com/votingworks/vxsuite/issues/5388

### PR DESCRIPTION
**Pull request by jonahkagan**
## Overview

Closes #5353 

Redesigns the smart card programming flow to make it more clear what the user should do at each step. Users were getting stuck on the initial screen, unsure what to do.

The general interaction model is the same (insert a card, view its details, change its programming). The main changes:
- Instead of opening a full-screen modal when inserting a card, show an illustration of the currently inserted card
- Add confirmation prompts for system admin card programming (rather than having a separate screen you have to navigate to for sys admin cards)
- Use callouts for status messages (better fitting with our design system)

## Demo Video or Screenshot
Programming blank cards and resetting PINs

https://github.com/user-attachments/assets/ad7157b9-ee67-487c-8cec-b2c74ab42a69

Unconfigured machine - can only program sys admin cards

https://github.com/user-attachments/assets/06b0cd18-fa18-4d89-8662-23ee1a6d5633

Reprogramming a card from a different election

https://github.com/user-attachments/assets/9bfca0b5-84a4-40d3-9e7b-57f7d2487c0e

Error handling

https://github.com/user-attachments/assets/0301ca48-fa63-4bb7-801c-accc7291d7f3

## Testing Plan
- Manual testing
- Updated automated tests

## Checklist

- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [x] I have added a screenshot and/or video to this PR to demo the change
- [x] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

**Comment by kshen0** [comment]
nit: Is the box on the left with rounded corners the card illustration? I think because the initial state is not an illustration of a card, I was primed to not think of that box as an illustration. As a stretch we might consider adding an svg of the card chip, but probably not worth it atm.

**Comment by jonahkagan** [comment]
> As a stretch we might consider adding an svg of the card chip

I played around with this a little bit while I was developing, but I couldn't get it to look right quickly so I punted. I agree with you that it might help though.

**Comment by jonahkagan** [comment]
Here's what that looks like. What do you think?
Screenshot-VxAdmin-2024-09-23T23:10:39 678Z

**Comment by kshen0** [comment]
> Here's what that looks like. What do you think?

I think it looks pretty good. It gets the point across clearly and the off-center position is even right. My main concern is IMHO it's a more sketch-like style than the high-fidelity vx logo, "insert card" instruction svg, etc. I think you were right to punt until we have time to design the card illustration more holistically. A complete pass might look like
* adding a brass color to the chip icon
* adding the black background bands/circles on the upper half of the card
* positioning the text exactly where it is on the physical cards

Definitely not a priority right now!

**Comment by jonahkagan** [comment]
I'm hesitant to try to replicate the way we physically style the cards, since we can easily change those stickers but won't be able to easily change this illustration to match. So I would say that a more abstracted style is probably the way to go if we do decide to do this.

**Comment by jonahkagan** [comment]
Thinking about it more, the illustration is growing on me. I think I want to run with it unless you object.

**Comment by kshen0** [comment]
> Thinking about it more, the illustration is growing on me. I think I want to run with it unless you object.

No objections, go for it

File: apps/admin/frontend/src/screens/smart_cards_screen.tsx [hunk 1]
```diff
@@ -0,0 +1,609 @@
...
+            {confirmLabel}
+          </Button>
+          <Button onPress={onClose}>Cancel</Button>
+        </React.Fragment>
+      }
+    />
+  );
+}
+
+function ProgramCard({
```
Comment (kshen0) [inline apps/admin/frontend/src/screens/smart_cards_screen.tsx:357]: Consider `ProgramCardInstructions`, `ProgramCardContainer`, etc. for same reasons as above

**Review by kshen0 — APPROVED** [review]

File: apps/admin/frontend/src/screens/smart_cards_screen.tsx [hunk 2]
```diff
@@ -0,0 +1,609 @@
...
+
+function checkDoesCardElectionMatchMachineElection(
+  programmedUser: User,
+  electionDefinition: ElectionDefinition
+): boolean {
+  if (!('electionKey' in programmedUser)) {
+    return false;
+  }
+  return deepEqual(
+    programmedUser.electionKey,
```
[the remark this answers is not in the document]
Comment (jonahkagan) [reply apps/admin/frontend/src/screens/smart_cards_screen.tsx:153]: This was copy-pasted from previous code, but it's a bit confusing. The check on L149 is narrowing the type of `programmedUser` to election manager or poll worker roles, which are the only ones that have `electionKey`. I will refactor to clarify this

File: apps/admin/frontend/src/screens/smart_cards_screen.tsx [hunk 3]
```diff
@@ -0,0 +1,609 @@
...
+    assert(role !== 'vendor');
+    programCardMutation.mutate(
+      { userRole: role },
+      {
+        onSuccess: (result) => {
+          setActionResult({
+            action: 'PinReset',
+            newPin: result.ok()?.pin,
+            role,
+            status: result.isOk() ? 'Success' : 'Error',
```
[the remark this answers is not in the document]
Comment (jonahkagan) [reply apps/admin/frontend/src/screens/smart_cards_screen.tsx:412]: This also was copy-pasted from previous code. The strings are type-checked via the `SmartCardActionResult` type, so I feel ok about it.

**Pull request merged**